src/widgets/AddMainTask/AddMainTask.tsx

```diff
@@ -9,73 +9,50 @@ import { useTranslation } from 'react-i18next';
 import { FormValidator } from 'shared/lib/helpers';
 
 import { AddMainTaskStyle as styles } from './AddMainTask.style';
+import { AddTask } from '../../features/Addtask/AddTask';
 
 export const AddMainTask = memo(() => {
     const dispatch = useAppDispatch();
     const { t } = useTranslation('translation');
     const email = useAppSelector(getUserData)?.email || '';
 
-    const [taskName, setTaskName] = useState<string>('');
-    const [taskNameError, setTaskNameError] = useState<string>('');
     const [performer, setPerformer] = useState<string>('');
     const [performerError, setPerformerError] = useState<string>('');
     const [isPrivate, setIsPrivate] = useState<boolean>(false);
     const [isImportant, setIsImportant] = useState<boolean>(false);
 
-    const handleChangeTaskName = (event: ChangeEvent<HTMLInputElement>) => {
-        if (taskNameError) setTaskNameError('');
-        setTaskName(event.target.value);
-    };
-
     const handleChangePerformer = (event: ChangeEvent<HTMLInputElement>) => {
         if (performerError) setPerformerError('');
         setPerformer(event.target.value);
     };
 
     const handleChangeIsPrivate = () => {
-        setIsPrivate((prevState) => !prevState);
+        setIsPrivate(!isPrivate);
     };
 
     const handleChangeIsImportant = () => {
-        setIsImportant((prevState) => !prevState);
+        setIsImportant(!isImportant);
     };
 
-    const createNewTask = () => {
-        const isTaskValid = FormValidator.isNotEmpty(taskName, setTaskNameError);
-        const isPerformerValid = FormValidator.performerChecking(performer, setPerformerError);
-        if (!isPerformerValid || !isTaskValid) return;
+    const createNewTask = (taskName: string) => {
+        const isPerformerValid = FormValidator.emailChecking(performer, setPerformerError, true);
+        if (!isPerformerValid) return false;
+
+        if (!taskName.trim()) return false;
 
         const task = createTaskDto(taskName, email, performer, isPrivate, isImportant);
         dispatch(createTask(task));
 
         setPerformer('');
         setIsImportant(false);
         setIsPrivate(false);
-        setTaskName('');
+
+        return true;
     };
```

Comment:
1. this can be done without returning a value
2. validate performer separately (`onBlur={handlePerformerValidation}`)